### PR TITLE
Add JDK 9 to travis build matrix

### PR DESCRIPTION
**Pull request by JLLeitschuh**
### Context

Related/demonstration of to #454

### Contributor Checklist
- [x] Base the PR against the `develop` branch
- [x] [Sign Gradle CLA](http://gradle.org/contributor-license-agreement/)
- [x] Provide integration tests to verify changes from a user perspective
- [x] Provide unit tests to verify logic
- [ ] ~Ensure that tests pass locally: `./gradlew check --parallel`~ (This is the problem I'm trying to demonstrate)

### CODE REVIEW

**Comment by JLLeitschuh** [comment]
@bamboo or @eskatos Did you disable showing the build status in PR's? Did you disable something in Travis itself? Why am I not seeing a build status showup in github?

**Comment by eskatos** [comment]
@JLLeitschuh the build choke on a very simple thing with Java 9, I pushed 2f40315d02667a0c930264f07df07bb15983d9c2 to `develop` to fix this, you may want to rebase your branch on top of `develop` so we'll see what comes next.

**Comment by JLLeitschuh** [comment]
Done! Lets see what happens!

**Comment by JLLeitschuh** [comment]
New log posted here:
https://github.com/gradle/kotlin-dsl/issues/454#issuecomment-321321277

**Comment by eskatos** [comment]
Ok, so now we're down to the Kotlin compiler failing:

```
java.lang.Exception: Error occurred in server thread; nested exception is: 
	java.lang.NoClassDefFoundError: com/sun/xml/internal/messaging/saaj/util/ByteOutputStream
	at org.jetbrains.kotlin.daemon.common.CompileService$CallResult$Error.get(CompileService.kt:60)
```

This issue states that it should be fixed by Kotlin 1.1.4: [KT-14988](https://youtrack.jetbrains.com/issue/KT-14988) Support running the Kotlin compiler on Java 9

See #414

**Comment by hastebrot** [comment]
Kotlin 1.1.4 was released today with the fix for Java 9

**Comment by bamboo** [comment]
Hey @JLLeitschuh, can you please rebase on top of `develop` so we can see how far the build goes now with Kotlin 1.1.4?

**Comment by JLLeitschuh** [comment]
Absolutely!!!! Woot!!!!

**Comment by JLLeitschuh** [comment]
Updated!

**Comment by JLLeitschuh** [comment]
New exception:
```
e: /home/travis/build/JLLeitschuh/kotlin-dsl/build.gradle.kts:13:17: Cannot access class 'java.net.URI'. Check your module classpath for missing or conflicting dependencies
e: /home/travis/build/JLLeitschuh/kotlin-dsl/build.gradle.kts:13:17: Cannot access class 'java.net.URI'. Check your module classpath for missing or conflicting dependencies
e: /home/travis/build/JLLeitschuh/kotlin-dsl/build.gradle.kts:13:23: Cannot access class 'java.net.URI'. Check your module classpath for missing or conflicting dependencies
e: Supertypes of the following classes cannot be resolved. Please make sure you have the required dependencies in the classpath:
    class org.gradle.api.Project, unresolved supertypes: Comparable<Project>
    class org.gradle.api.plugins.ExtensionAware, unresolved supertypes: Object
    class org.gradle.api.plugins.PluginAware, unresolved supertypes: Object
    class groovy.lang.Closure, unresolved supertypes: Cloneable, Runnable, Serializable
    class groovy.lang.GroovyObjectSupport, unresolved supertypes: Object
    class groovy.lang.GroovyObject, unresolved supertypes: Object
    class groovy.lang.GroovyCallable, unresolved supertypes: Callable<V>
    class org.gradle.api.Action, unresolved supertypes: Object
    class org.gradle.api.NamedDomainObjectFactory, unresolved supertypes: Object
    class org.gradle.api.initialization.dsl.ScriptHandler, unresolved supertypes: Object
    class org.gradle.api.plugins.ExtraPropertiesExtension, unresolved supertypes: Object
    class org.gradle.api.NamedDomainObjectList, unresolved supertypes: List<T>
    class org.gradle.api.DomainObjectCollection, unresolved supertypes: Collection<T>
    class org.gradle.util.Configurable, unresolved supertypes: Object
    class org.gradle.api.artifacts.repositories.ArtifactRepository, unresolved supertypes: Object
    class org.gradle.api.artifacts.repositories.AuthenticationSupported, unresolved supertypes: Object
    class org.gradle.api.artifacts.dsl.DependencyHandler, unresolved supertypes: Object
```

**Comment by bamboo** [comment]
These messages seem to indicate the JDK classpath is not being properly recognized:

> class org.gradle.api.Action, unresolved supertypes: Object

@ligee, any insight here?

**Comment by JLLeitschuh** [comment]
Do you need to add a `module-info.java` file somewhere? Is that what the kotlin compiler is looking for?
https://blog.jetbrains.com/kotlin/2017/08/kotlin-1-1-4-is-out/

**Comment by JLLeitschuh** [comment]
Just rebased my changes on develop. Guessing that nothing will have changed.

**Comment by bamboo** [comment]
Hey, @JLLeitschuh, please rebase it on top of `develop` as it now contains a Gradle distro with the `java.home` workaround.

**Comment by JLLeitschuh** [comment]
Rebased 🤞 hopefully this works!

**Comment by JLLeitschuh** [comment]
I'm not able to reproduce these travis failures on my machine running under JDK 9. @bamboo Thoughts? I can't even figure out what test has failed specifically. Maybe someone should try re-running the build?

There are a bunch of tests that are supposed to fail in there but their stack traces obscure the ones that are true failures from the ones that are not.

File: fixtures/build.gradle.kts [hunk 1]
```diff
@@ -3,7 +3,14 @@ import org.gradle.api.tasks.GradleBuild
...
 
-val subProjectTasks = rootDir.listFiles().filter { isProjectDir(it) }.map { subProjectDir ->
+fun isCompatibleWithJDK(candidate: File) =
+    candidate.name.endsWith("kotlin-1.0") && JavaVersion.current() < JavaVersion.VERSION_1_9
+
+val subProjectTasks = rootDir
+    .listFiles()
+    .filter { isProjectDir(it) }
+    .filter { isCompatibleWithJDK(it) }
+    .map { subProjectDir ->
```
Comment (bamboo) [inline fixtures/build.gradle.kts]: :x: Indentation looks off here.

Comment (JLLeitschuh) [reply fixtures/build.gradle.kts]: Should be resolved now.

**Comment by bamboo** [comment]
Thank you very much, @JLLeitschuh!

**Pull request merged**